Tools/cases_generator/parser.py

```diff
@@ -62,8 +62,17 @@ class Block(Node):
 @dataclass
 class StackEffect(Node):
     name: str
-    type: str = ""
-    # TODO: array, condition
+    type: str = ""  # Optional `:type`
+    size: str = ""  # Optional `[size]`
+    # Note: we can have type or size but not both
+    # TODO: condition (can be combined with type but not size)
+
+
+@dataclass
+class Dimension(Node):
+    # NAME ['*' NUMBER]
+    name: str
+    size: int = 1
```

Comment:
Again, maybe consider something like:

```suggestion
    # NAME ['*' NUMBER] [('+' | '-') NUMBER]
    name: str
    scale: int = 1
    shift: int = 0
```